### PR TITLE
feat: add no-unsafe-takeuntil

### PR DESCRIPTION
**Pull request by timdeschryver**
Part of #1 

I don't know why but it feels like I missed something here...

TSLint:

- https://github.com/cartant/rxjs-tslint-rules/blob/master/source/rules/rxjsNoUnsafeTakeuntilRule.ts
- https://github.com/cartant/rxjs-tslint-rules/blob/master/test/v6/fixtures/no-unsafe-takeuntil/default/fixture.ts.lint
- https://github.com/cartant/rxjs-tslint-rules/blob/master/test/v6/fixtures/no-unsafe-takeuntil/allow-custom/fixture.ts.lint
- https://github.com/cartant/rxjs-tslint-rules/blob/master/test/v6/fixtures/no-unsafe-takeuntil/allow-default/fixture.ts.lint

### CODE REVIEW

File: source/rules/no-unsafe-takeuntil.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    docs: {
+      category: "RxJS",
+      description: "Disallows the application of operators after takeUntil.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Applying operators after `takeUntil` are forbidden."
+    },
+    schema: [
```
Comment (timdeschryver) [inline source/rules/no-unsafe-takeuntil.ts]: This rule, `no-exposed-subjects`, and `no-sharereplay` all have options and I noticed tslint has an option description but these converted rules are missing the description. I can add them if you want, but we might want to use something different as `Lint.Utils.dedent`, maybe move `common-tags` to dependencies?

File: tests/rules/no-unsafe-takeuntil.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    {
+      code: stripIndent`
+        import { of } from "rxjs";
+        import { switchMap, takeUntil } from "rxjs/operators";
+
+        const a = of("a");
+        const b = of("b");
+        const c = of("d");
+
+        const d = a.pipe(switchMap(_ => b), pipe(takeUntil(d), switchMap(_ => b))).subscribe();
```
Comment (timdeschryver) [inline tests/rules/no-unsafe-takeuntil.ts]: I think we might want to see this is an error?
rxjs-tslint-rules doesn't see this an error, so for now I left it as is.

File: tests/rules/no-unsafe-takeuntil.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+          messageId: "forbidden",
+          line: 8,
+          column: 18,
+          endLine: 8,
+          endColumn: 27
+        }
+      ]
+    },
+    {
+      code: stripIndent`
```
Comment (timdeschryver) [inline tests/rules/no-unsafe-takeuntil.ts]: Ported over from https://github.com/cartant/rxjs-tslint-rules/blob/master/test/v6/fixtures/issues/49/fixture.ts.lint.

Do you want to have these issues grouped as before?

File: source/rules/no-unsafe-takeuntil.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        if (!isReferenceType(node) || !couldBeObservable(node)) {
+          return;
+        }
+
+        node.arguments.reduceRight((isError, arg) => {
+          if (!isCallExpression(arg)) {
+            return true;
+          }
+
+          if (!isIdentifier(arg.callee)) {
```
Comment (timdeschryver) [inline source/rules/no-unsafe-takeuntil.ts]: For safety, AFAIK this should always be an Identifier

File: source/rules/no-unsafe-takeuntil.ts (hunk 1 above)
Comment (cartant) [reply source/rules/no-unsafe-takeuntil.ts]: > maybe move `common-tags` to dependencies?

Yep.

I most likely copied rule metadata from someplace else that didn't have a description. If there is a description property that's specific to the options, we should fill it out.

File: tests/rules/no-unsafe-takeuntil.ts (hunk 2 above)
Comment (cartant) [reply tests/rules/no-unsafe-takeuntil.ts]: I don't mind. This is an edge case that's unlikely to occur. It's in the tests, IIRC, to show that the rule's implementation does not care about this usage.

File: tests/rules/no-unsafe-takeuntil.ts (hunk 3 above)
Comment (cartant) [reply tests/rules/no-unsafe-takeuntil.ts]: I guess that keeping some reference to the original issue might be useful context, but other than that, I don't mind how they are grouped.

File: source/rules/no-unsafe-takeuntil.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          !node.arguments ||
+          !isReferenceType(node) ||
+          !couldBeObservable(node)
+        ) {
+          return;
+        }
+
+        node.arguments.reduceRight((isError, arg) => {
+          if (!isCallExpression(arg)) {
+            return true;
```
Comment (cartant) [inline source/rules/no-unsafe-takeuntil.ts]: This should not set the error condition. It's possible to pass something other than a `CallExpression` as an argument to `pipe`. E.g. you could pass `source => source` or a variable, etc. Operators return functions that take an observable and return an observable.
